tests/components/watergate/test_valve.py

```diff
@@ -0,0 +1,49 @@
+"""Tests for the Watergate coordinator."""
+
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.valve import DOMAIN as VALVE_DOMAIN, ValveState
+from homeassistant.const import ATTR_ENTITY_ID, SERVICE_CLOSE_VALVE, SERVICE_OPEN_VALVE
+from homeassistant.core import HomeAssistant
+
+
+async def test_change_valve_state(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_watergate_client,
+    mock_entry,
+) -> None:
+    """Test entities become unavailable after failed update."""
+    mock_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_entry.entry_id)
+    await hass.async_block_till_done()
+    entity_id = "valve.sonic"
+
+    current_state = hass.states.get(entity_id).state
+    assert current_state
+    assert current_state == ValveState.OPEN
+    assert current_state == snapshot
+
+    await hass.services.async_call(
+        VALVE_DOMAIN,
+        SERVICE_CLOSE_VALVE,
+        {ATTR_ENTITY_ID: entity_id},
+        blocking=True,
+    )
+
+    current_state = hass.states.get(entity_id).state
+    assert current_state
+    assert current_state == ValveState.CLOSING
```

Comment:
```suggestion
    current_state = hass.states.get(entity_id)
    assert current_state
    assert current_state.state == ValveState.CLOSING
```
for the other occurrences as well. This is to ensure, you'll at least have the entity, before getting it's `state`

Reply:
Got it. But I will do it only once in one test on first occurence, than I will be checking only test.